nav2_behavior_tree/include/nav2_behavior_tree/plugins/decorator/is_path_longer_on_approach.hpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2022 Neobotix GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__DECORATOR__IS_PATH_LONGER_ON_APPROACH_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__DECORATOR__IS_PATH_LONGER_ON_APPROACH_HPP_
+
+#include <string>
+#include <memory>
+#include <limits>
+
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "behaviortree_cpp_v3/decorator_node.h"
+
+namespace nav2_behavior_tree
+{
+
+/**
+ * @brief A BT::DecoratorNode that ticks its child everytime when the length of
+ * the new path is smaller than the old one by the length given by the user.
+ */
+class IsPathLongerOnApproach : public BT::DecoratorNode
+{
+public:
+  /**
+   * @brief A constructor for nav2_behavior_tree::IsPathLongerOnApproach
+   * @param name Name for the XML tag for this node
+   * @param conf BT node configuration
+   */
+  IsPathLongerOnApproach(
+    const std::string & name,
+    const BT::NodeConfiguration & conf);
+
+  /**
+   * @brief Creates list of BT ports
+   * @return BT::PortsList Containing node-specific ports
+   */
+  static BT::PortsList providedPorts()
+  {
+    return {
+      BT::InputPort<nav_msgs::msg::Path>("path", "Path to follow"),
+      BT::InputPort<double>(
+        "prox_leng", 1.0,
```

Comment:
I'd think we'd want to apply this far before 1m away. Maybe 5m? What do you think is a good default?

Also the description isn't very clear, I'd do something like "Integrated path length proximity to the goal pose to apply the behavior (m)"

Reply:
anywhere around 3-5m will be a good choice. So, as a default 3m sounds good.